### PR TITLE
Add checks workflow

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation

This PR introduces a CI job to check code formats. It follows the [checks](https://github.com/optuna/optuna/blob/master/.github/workflows/checks.yml) job of [optuna/optuna](https://github.com/optuna/optuna) except for `mypy`. This is because examples do not have type hinting.

## Changes

It adds `black`, `flake8` and ~`isort`~.
`isort` is not applied to the examples in `optuna/optuna`. So, I removed it. We may introduce it in a follow-up PR.

### CODE REVIEW

**Comment by hvy** [comment]
Should it be `.github/workflow/checks.yml` -> `.github/workflows/checks.yml` ? Noticed it's not kicked.

**Comment by hvy** [comment]
Do you think we should apply the same configurations as in `optuna/optuna` (pyproject.toml, setup.cfg)? I don't have a strong opinion but that'll probably make maintenance for us easier, rather than having two policies.

**Comment by toshihikoyanase** [comment]
> Should it be `.github/workflow/checks.yml` -> `.github/workflows/checks.yml` ?

Thanks. I renamed it.

> Do you think we should apply the same configurations as in optuna/optuna (pyproject.toml, setup.cfg)?

You're right. I just forgot to add them.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  | _build
+  | buck-out
+  | build
+  | dist
+  | docs
+)/
+'''
+
+[tool.pytest.ini_options]
+addopts = "--color=yes"
```
Comment (hvy) [inline pyproject.toml:21]: Are these lines used?
```suggestion
```

Comment (toshihikoyanase) [reply pyproject.toml]: No, they aren't. Thanks.

**Review by hvy — APPROVED** [review]
LGTM!

**Pull request merged**